### DISCUSSION TITLE
How to Rerun failed scenario automatically using cucumber rerun

### DISCUSSION THREAD

**Discussion by hunterabhi212223** [cucumber/common#2175] [Q&A Java]
As  we know when any feature file failed we are storing in a text file. To run that failed.txt file we are using 2nd runner file. If we run second runner file it is executing the failed one only.
Is there any way after completion of first runner it will trigger second runner file to run failed one automatically

**Comment by luke-hill** [comment]
Which flavour of cucumber. There is the inbuilt retry logic in some flavours of cucumber

**Reply by hunterabhi212223** [reply]
// Runner1.java
@RunWith(Cucumber.class)
@CucumberOptions(
    features = "src/test/resources/features",
    glue = "stepDefinitions",
    tags = "@FirstSet",
    plugin = {"pretty", "html:target/cucumber-reports", "rerun:target/rerun1.txt"}
)
public class Runner1 {
}

// RerunFailedTests1.java
@RunWith(Cucumber.class)
@CucumberOptions(
    features = "@target/rerun1.txt",
    glue = "stepDefinitions",
    plugin = {"pretty", "html:target/cucumber-reports-rerun"}
)
public class RerunFailedTests1 {
}

**Comment by mpkorstanje** [comment]
It depends.

If you're running your classes through and IDE it is not possible. Unless you can configure your IDE to run multiple classes.

Otherwise you can configure Maven or Gradle with multiple test executions. Or with a single execution where the runner runs before the rerunner.

Finally, if you want to have full control, you could use the JUnit 5 Platform Launcher API in combination with the Cucumber JUnit Platform Engine to execute your test programmatically.

https://github.com/cucumber/cucumber-jvm/tree/main/cucumber-junit-platform-engine#using-the-junit-platform-launcher-api

**Reply by hunterabhi212223** [reply]
It wont support.I am using junit-jupiter-api: 5.5.2 and junit 4.8.0

I tried with gradle retry plugin. gradle is not supporting junit platform()
Could you suggest alternate approach for running runner file(having failure.txt file) either in gradle or any other

**Reply by mpkorstanje** [reply]
I would suggest upgrading. JUnit 4 is getting rather old.

But gradle is pretty good at running arbitrary code during the build. So you could always use `io.cucumber.core.cli.Main` for the run and the rerun. Though I'm not that much into Gradle so you'll have to work out the specifics yourself.
